src/oph/heratepalvelu/db/dynamodb.clj

```diff
@@ -194,6 +194,46 @@
          items (.items response)]
      (vec (map map-attribute-values-to-vals items)))))
 
+(defn query-items-with-expression
+  "Hakee tietueita tiettyjen ehtojen perusteella. Funktion key-expr
+  -argumentti seuraa expression-syntaksia.
+
+  Objektista options voi löytyä myös suodatusekspressio ja/tai muita ehdollisia
+  parametreja:
+    :index               - index, jota käytetään key-condsissa
+    :limit               - maksimimäärä palautettavia tietueita
+    :filter-expression   - DynamoDB-formaatissa oleva predikaatti, jonka
+                           perusteella palautettavat tietueet suodatetaan
+    :expr-attr-names     - lista avaimista ja kentännimiä, joilla avaimet
+                           korvataan filter-expressionissa
+    :expr-attr-vals      - lista avaimista ja attribuuttiarvoista, jotka
+                           laitetaan avaimien sijoihin, kun ekspressio
+                           evaluoidaan"
+  ([key-expr options]
+   (query-items-with-expression key-expr options (:herate-table env)))
+  ([key-expr options table]
+   (let [response (.query ddb-client (-> (QueryRequest/builder)
+                                         (.tableName table)
+                                         (.keyConditionExpression key-expr)
```

Comment:
Jos tarkoitus on siis korvata (vanhentunut?) `.keyConditions` uudella `.keyConditionExpression`:lla, niin mun mielestä siistein tapa olis muuttaa `query-items` eikä tehdä sille rinnakkaista uutta kirjastokutsua (eihän meidän oo tarkoitus jättää vanhentuneita tapoja käyttää APIa koodikantaan).  Musta vaikuttaa siltä, että query-items:n saamasta tietorakenteesta pystyy rakentelemaan myös `.keyConditionExpression`:lle lausekkeen.  Eli query-items pystyis näyttämään ulospäin samalta kuin ennenkin, mutta käyttämään uudempaa APIa.

Reply:
Sopiiko että tiketöin nuo pyydetyt asiat uuteen tikettiin? Tämän koodimuutoksen oli pointtina haarukoida, että onko siitä apua tämän kyseisen tiketin ongelman ratkaisussa.

Reply:
Jes, tiketöidään kunnollisempi korjaus.